host/src/peripheral.rs

```diff
@@ -106,6 +106,38 @@ impl<'d, C: Controller> Peripheral<'d, C> {
         })
     }
 
+    /// Update the advertisment adv_data and/or scan_data. Does not change any
+    /// other advertising parameters. If no advertising is active, this will not
+    /// produce any observable effect. This is typically useful when
+    /// implementing a BLE beacon that only broadcasts advertisement data and
+    /// does not accept any connections.
+    pub async fn update_adv_data<'k>(&mut self, data: Advertisement<'k>) -> Result<(), BleHostError<C::Error>>
+    where
+        C: for<'t> ControllerCmdSync<LeSetAdvData>
+            + ControllerCmdSync<LeSetAdvParams>
+            + for<'t> ControllerCmdSync<LeSetAdvEnable>
+            + for<'t> ControllerCmdSync<LeSetScanResponseData>,
```

Comment:
I don't see any reason why you need to have constraints other than LeSetScanResponseData and LeSetAdvData here.